src/actions/mover.ts

```diff
@@ -0,0 +1,339 @@
+        createSerializedKey(KeyCodes.LEFT, [KeyCodes.ALT]),
+        createSerializedKey(KeyCodes.LEFT, [KeyCodes.CTRL]),
+      ],
+    },
+    {
+      name: 'Move right, unconstraind',
+      preconditionFn: (workspace) => this.isMoving(workspace),
+      callback: (workspace) => this.moveUnconstrained(workspace, 1, 0),
+      keyCodes: [
+        createSerializedKey(KeyCodes.RIGHT, [KeyCodes.ALT]),
+        createSerializedKey(KeyCodes.RIGHT, [KeyCodes.CTRL]),
+      ],
+    },
+    {
+      name: 'Move up unconstrained',
+      preconditionFn: (workspace) => this.isMoving(workspace),
+      callback: (workspace) => this.moveUnconstrained(workspace, 0, -1),
+      keyCodes: [
+        createSerializedKey(KeyCodes.UP, [KeyCodes.ALT]),
+        createSerializedKey(KeyCodes.UP, [KeyCodes.CTRL]),
+      ],
+    },
+    {
+      name: 'Move down, unconstrained',
+      preconditionFn: (workspace) => this.isMoving(workspace),
+      callback: (workspace) => this.moveUnconstrained(workspace, 0, 1),
+      keyCodes: [
+        createSerializedKey(KeyCodes.DOWN, [KeyCodes.ALT]),
+        createSerializedKey(KeyCodes.DOWN, [KeyCodes.CTRL]),
+      ],
+    },
+  ];
+
+  menuItems: ContextMenuRegistry.RegistryItem[] = [
+    {
+      displayText: 'Move Block (M)',
+      preconditionFn: (scope) => {
+        const workspace = scope.block?.workspace as WorkspaceSvg | null;
+        if (!workspace) return 'hidden';
+        return this.canMove(workspace, scope.block) ? 'enabled' : 'disabled';
+      },
+      callback: (scope) => {
+        const workspace = scope.block?.workspace as WorkspaceSvg | null;
+        if (!workspace) return false;
+        this.startMove(workspace, scope.block);
+      },
+      scopeType: ContextMenuRegistry.ScopeType.BLOCK,
+      id: 'move',
+      weight: 8.5,
+    },
+  ];
+
+  /**
+   * Install the actions as both keyboard shortcuts and (where
+   * applicable) context menu items.
+   */
+  install() {
+    for (const shortcut of this.shortcuts) {
+      ShortcutRegistry.registry.register(shortcut);
+    }
+    for (const menuItem of this.menuItems) {
+      ContextMenuRegistry.registry.register(menuItem);
+    }
+  }
+
+  /**
+   * Uninstall these actions.
+   */
+  uninstall() {
+    for (const shortcut of this.shortcuts) {
+      ShortcutRegistry.registry.unregister(shortcut.name);
+    }
+    for (const menuItem of this.menuItems) {
+      ContextMenuRegistry.registry.unregister(menuItem.id);
+    }
+  }
+
+  /**
+   * Returns true iff we are able to begin moving the given block (or,
+   * if no block supplied, the block wich currently has focus) on the
+   * given workspace.
+   *
+   * @param workspace The workspace to move on.
+   * @returns True iff we can beign a move.
+   */
+  canMove(workspace: WorkspaceSvg, block?: Block) {
+    if (!block) {
+      const cursor = workspace?.getCursor();
+      const curNode = cursor?.getCurNode();
+      block = curNode?.getSourceBlock() ?? undefined;
+    }
```

Comment:
Might be worth factoring this out into a `getCurrentBlock()` since `startMove` also appears to need it